app/src/controllers/organization/sagas.js

```diff
@@ -58,10 +64,43 @@ function* watchFetchOrganizationBySlug() {
   yield takeEvery(FETCH_ORGANIZATION_BY_SLUG, fetchOrganizationBySlug);
 }
 
+function* fetchOrganizationSettings({ payload: organizationId }) {
+  yield put(
+    fetchDataAction(FETCH_ORGANIZATION_SETTINGS)(URLS.organizationSettings(organizationId)),
+  );
+}
+
+function* watchFetchOrganizationSettings() {
+  yield takeEvery(FETCH_ORGANIZATION_SETTINGS, fetchOrganizationSettings);
+}
+
+function* prepareActiveOrganizationSettings({ payload: { organizationSlug } }) {
+  let activeOrganization = yield select(activeOrganizationSelector);
+  try {
+    if (!activeOrganization || organizationSlug !== activeOrganization?.slug) {
+      yield take(createFetchPredicate(FETCH_ORGANIZATION_BY_SLUG));
+      activeOrganization = yield select(activeOrganizationSelector);
+    }
+    yield put(fetchOrganizationSettingsAction(activeOrganization.id));
```

Comment:
No need to dispatch a redux action which will immediately dispatch another redux action.
You can use `call` to execute saga directly.
```suggestion
    yield call(fetchOrganizationSettings, { payload: activeOrganization.id });
```

Reply:
Yes, thank you! Resolved in the last commit.